recipes/magithub

```diff
@@ -0,0 +1,4 @@
+(magithub :fetcher github
+          :repo "vermiculus/magithub"
+          :files ("lisp/magithub.el"
```

Comment:
You'd just use `(magithub :fetcher github :repo "vermiculus/magithub")` and omit the LICENSE -- we don't normally package those. I've added that recipe manually now, and closed this PR.

> I based the file itself heavily off of magit.el since I don't have the best track record when it comes to conforming to the emacs package ecosystem

`M-x auto-insert` is your friend. Run it in a fresh, empty emacs lisp file. :-)